File: 2017/02/day02_2.rb

def check_number(number, line)
  sum = 0
  line.each do |n|
    if number != n && number > n && number % n == 0
      sum += number / n
    end
  end
  return sum
end

def check_line(line)
  res = 0
  line.each do |number|
    res += check_number(number,line)
  end
  return res
end

sum = 0
file = File.new("input.txt", "r")
while (l = file.gets)
  line = Array.new
  l.strip.split(" ").each do |num|
    line.push(num.to_i)
  end
  sum += check_line(line)
end
file.close

puts sum

File: 2020/02/day02_2.py
passwords = []
file = open("input.txt", "r") 
for line in file:
	tmp = line.strip().split()
	
	password = tmp[0].split("-")
	passwords.append(password)

	password.append(tmp[1].strip(":"))
	password.append(tmp[2])

valid_passwords = 0
for i in passwords:
	pos1 = int(i[0])-1
	pos2 = int(i[1])-1
		
	if ( (i[3][pos1] == i[2]) ^ (i[3][pos2] == i[2]) ):
		valid_passwords += 1

print(valid_passwords)